src/openforms/forms/api/serializers/form.py

```diff
@@ -217,21 +269,87 @@ def create(self, validated_data):
+            warnings.warn(
+                (
+                    "Form 'registration_backend' and 'registration_backend_options' "
+                    "are deprecated, use 'registration_backends' instead."
+                ),
+                DeprecationWarning,
+            )
+            registration_backends.append(
+                {
+                    "key": "default",
+                    "name": _("Default"),
+                    "backend": validated_data.pop("registration_backend"),
+                    "options": registration_backend_options,
+                }
+            )
+
         instance = super().create(validated_data)
         ConfirmationEmailTemplate.objects.set_for_form(
             form=instance, data=confirmation_email_template
         )
+        FormRegistrationBackend.objects.bulk_create(
+            FormRegistrationBackend(form=instance, **backend)
+            for backend in registration_backends
+        )
         return instance
 
+    def _update_v2_registration_backend(self, form, validated_data):
+        new_data = {}
+        if "registration_backend" in validated_data:
+            new_data["backend"] = validated_data.pop("registration_backend")
+        if "registration_backend_options" in validated_data:
+            new_data["options"] = validated_data.pop("registration_backend_options")
+        if not any(v is not None for v in new_data.values()):
+            return
+
+        warnings.warn(
+            (
+                "Form 'registration_backend' and 'registration_backend_options' "
+                "are deprecated, use 'registration_backends' instead."
+            ),
+            DeprecationWarning,
+        )
+
+        backend = form.registration_backends.first() or FormRegistrationBackend(
+            form=form, name=_("Default"), key="default"
+        )
+        serializer = FormRegistrationBackendSerializer(
+            instance=backend, data=new_data, context=self.context, partial=self.partial
+        )
+        # validated_data ought to be correct already
+        assert serializer.is_valid()
+        serializer.save()
+
     @transaction.atomic()
     def update(self, instance, validated_data):
         confirmation_email_template = validated_data.pop(
             "confirmation_email_template", None
         )
+        registration_backends = validated_data.pop("registration_backends", None)
+
+        self._update_v2_registration_backend(instance, validated_data)
+
         instance = super().update(instance, validated_data)
         ConfirmationEmailTemplate.objects.set_for_form(
             form=instance, data=confirmation_email_template
         )
+        if registration_backends is None:
+            return instance
+
+        instance.registration_backends.all().delete()
+        FormRegistrationBackend.objects.bulk_create(
+            FormRegistrationBackend(form=instance, **backend)
+            for backend in registration_backends
+        )
```

Comment:
So edge case:

1. form with registration backend is configured
2. submission is created and completed for this registration backend
3. submission registration attempt fails
4. form is updated with new configuration (possibly to fix registration issue)
5. submission registration attempt retries

The FK from submission to registration backend is broken due to the `models.SET_NULL` off the top of my head - which makes the registration retry break as there is no longer any registration meta-information available.

I'm not sure what the best approach is here - maybe just store the `key` of the registration backend used? Since copying/freezing the information doesn't allow fixing the registration issues after the fact :thinking: